app/scripts/controllers/swaps.ts

```diff
@@ -50,29 +49,39 @@ import { Numeric } from '../../../shared/modules/Numeric';
 import { isEqualCaseInsensitive } from '../../../shared/modules/string-utils';
 import { isSwapsDefaultTokenAddress } from '../../../shared/modules/swaps.utils';
 import {
+  controllerName,
   FALLBACK_QUOTE_REFRESH_TIME,
   MAX_GAS_LIMIT,
   POLL_COUNT_LIMIT,
-  swapsControllerInitialState,
+  getDefaultSwapsControllerState,
 } from './swaps.constants';
+import {
+  calculateGasEstimateWithRefund,
+  getMedianEthValueQuote,
+} from './swaps.utils';
 import type {
   FetchTradesInfoParams,
   FetchTradesInfoParamsMetadata,
-  Quote,
-  QuoteSavings,
+  SwapsControllerMessenger,
   SwapsControllerOptions,
   SwapsControllerState,
-  SwapsControllerStore,
+  Quote,
+  QuoteSavings,
   Trade,
 } from './swaps.types';
-import {
-  calculateGasEstimateWithRefund,
-  getMedianEthValueQuote,
-} from './swaps.utils';
 
-export default class SwapsController {
-  public store: SwapsControllerStore;
+const metadata: StateMetadata<SwapsControllerState> = {
+  swapsState: {
+    persist: false,
+    anonymous: false,
+  },
+};
 
+export default class SwapsController extends BaseController<
```

Comment:
Probably out-of-scope for this PR, but since this controller contains polling logic, could it be refactored to inherit from `StaticIntervalPollingController` in the future?

Reply:
Thanks for the headsup! I didnt know we had that. I see that `TokenDetectionController` is using it this way also. Like you said, i consider this a out of scope for this PR, but will definitively will add this on our to-do list